### PR TITLE
[DOC] Enhanced RDoc for IO

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treated:
- #sync
- #sync=
- #fsync
- #fdatasync
- #fileno
- #pid
- #inspect
- #to_io

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -2457,15 +2457,16 @@ rb_io_eof(VALUE io)
...
  *  call-seq:
- *     ios.sync    -> true or false
+ *    sync -> true or false
  *
- *  Returns the current ``sync mode'' of <em>ios</em>. When sync mode is
- *  true, all output is immediately flushed to the underlying operating
- *  system and is not buffered by Ruby internally. See also
- *  IO#fsync.
+ *  Returns the value of the most recent call to IO#sync=,
+ *  or +false+ if the method has not been called; see IO#sync=:
```
Comment (jeremyevans) [inline io.c:2463]: The previous language was more accurate. There are cases where the sync mode is set without `IO#sync=` (such as on the write pipe returned by `IO.pipe`).  I would  continue to state that this returns the sync mode, even if you only define the sync mode in `IO#sync=`

File: io.c [hunk 2]
```diff
@@ -2630,22 +2650,23 @@ rb_io_descriptor(VALUE io)
 
 /*
  *  call-seq:
- *     ios.pid    -> integer
+ *    pid -> integer
```
Comment (jeremyevans) [inline io.c:2653]: `integer` or `nil`

Somewhere in the description, we should describe that this returns `nil` for instances not created by `IO.popen`.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good.  Only a few minor changes requested, please see inline comments.

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c (hunk 2 above)
Comment (BurdetteLamar) [reply io.c:2654]: Done.  I saw the nil in the previous doc, but could not figure out when it occurs.

Comment (jeremyevans) [reply io.c:2654]: I think the call-seq should still be updated to say `integer or nil`.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Please change the call-seq for #pid before committing.  After that, OK to commit without further review.

**Pull request merged**